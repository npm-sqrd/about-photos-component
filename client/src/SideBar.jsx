import React from 'react';
import './styles.css';

const SideBar = ({ price, hours, style, phone }) => {
  return (
    <div>
      <img src="https://s3-us-west-1.amazonaws.com/imagesfornamson/21.png" className="moneyIcon" />
      <div> 
        <div className="side">Price:</div>
        <div className="sideInfoPrice">{price}</div>
      </div>
      <div> 
        <div className="side">Hours of Operation:</div>
        <img src="https://s3-us-west-1.amazonaws.com/imagesfornamson/clock.png" className="clockIcon" />
        <div className="sideInfoHours">{hours}</div>
      </div>
      <div>
        <div className="side">Dress code:</div>
        <img src="https://s3-us-west-1.amazonaws.com/imagesfornamson/img_473372.png" className="dressIcon" />
        <div className="sideInfoStyle">{style}</div>
      </div>
      <div>
        <div className="side">Phone Number:</div>
        <img src="https://s3-us-west-1.amazonaws.com/imagesfornamson/telephone_318-61547.jpg" className="phoneIcon" />
        <div className="sideInfoPhone" >{phone}</div>
      </div>
    </div>
  );
};

export default SideBar;
