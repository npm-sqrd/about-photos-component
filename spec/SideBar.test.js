import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';

import SideBar from '../client/src/SideBar.jsx';

Enzyme.configure({ adapter: new Adapter() });

describe('About Component', () => {
  test('should render correctly', () => {
    expect(shallow(<SideBar />)).toMatchSnapshot();
  });
});

describe("rendered `SideBar`", () => {
  it("received four props", () => {
    const component = shallow(<SideBar />);
    expect(component.props().children.length).toBe(4);
  });
});
