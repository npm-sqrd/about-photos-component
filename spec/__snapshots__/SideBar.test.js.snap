// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SideBar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          className="moneyIcon"
          src="https://s3-us-west-1.amazonaws.com/imagesfornamson/21.png"
        />,
        <div>
          <div
            className="side"
          >
            Price:
          </div>
          <div
            className="sideInfoPrice"
          />
        </div>,
        <div>
          <div
            className="side"
          >
            Hours of Operation:
          </div>
          <img
            className="clockIcon"
            src="https://s3-us-west-1.amazonaws.com/imagesfornamson/clock.png"
          />
          <div
            className="sideInfoHours"
          />
        </div>,
        <div>
          <div
            className="side"
          >
            Dress code:
          </div>
          <img
            className="dressIcon"
            src="https://s3-us-west-1.amazonaws.com/imagesfornamson/img_473372.png"
          />
          <div
            className="sideInfoStyle"
          />
        </div>,
        <div>
          <div
            className="side"
          >
            Phone Number:
          </div>
          <img
            className="phoneIcon"
            src="https://s3-us-west-1.amazonaws.com/imagesfornamson/telephone_318-61547.jpg"
          />
          <div
            className="sideInfoPhone"
          />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "moneyIcon",
          "src": "https://s3-us-west-1.amazonaws.com/imagesfornamson/21.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="side"
            >
              Price:
            </div>,
            <div
              className="sideInfoPrice"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Price:",
              "className": "side",
            },
            "ref": null,
            "rendered": "Price:",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "sideInfoPrice",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="side"
            >
              Hours of Operation:
            </div>,
            <img
              className="clockIcon"
              src="https://s3-us-west-1.amazonaws.com/imagesfornamson/clock.png"
            />,
            <div
              className="sideInfoHours"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Hours of Operation:",
              "className": "side",
            },
            "ref": null,
            "rendered": "Hours of Operation:",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "clockIcon",
              "src": "https://s3-us-west-1.amazonaws.com/imagesfornamson/clock.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "sideInfoHours",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="side"
            >
              Dress code:
            </div>,
            <img
              className="dressIcon"
              src="https://s3-us-west-1.amazonaws.com/imagesfornamson/img_473372.png"
            />,
            <div
              className="sideInfoStyle"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Dress code:",
              "className": "side",
            },
            "ref": null,
            "rendered": "Dress code:",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "dressIcon",
              "src": "https://s3-us-west-1.amazonaws.com/imagesfornamson/img_473372.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "sideInfoStyle",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="side"
            >
              Phone Number:
            </div>,
            <img
              className="phoneIcon"
              src="https://s3-us-west-1.amazonaws.com/imagesfornamson/telephone_318-61547.jpg"
            />,
            <div
              className="sideInfoPhone"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Phone Number:",
              "className": "side",
            },
            "ref": null,
            "rendered": "Phone Number:",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "phoneIcon",
              "src": "https://s3-us-west-1.amazonaws.com/imagesfornamson/telephone_318-61547.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "sideInfoPhone",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            className="moneyIcon"
            src="https://s3-us-west-1.amazonaws.com/imagesfornamson/21.png"
          />,
          <div>
            <div
              className="side"
            >
              Price:
            </div>
            <div
              className="sideInfoPrice"
            />
          </div>,
          <div>
            <div
              className="side"
            >
              Hours of Operation:
            </div>
            <img
              className="clockIcon"
              src="https://s3-us-west-1.amazonaws.com/imagesfornamson/clock.png"
            />
            <div
              className="sideInfoHours"
            />
          </div>,
          <div>
            <div
              className="side"
            >
              Dress code:
            </div>
            <img
              className="dressIcon"
              src="https://s3-us-west-1.amazonaws.com/imagesfornamson/img_473372.png"
            />
            <div
              className="sideInfoStyle"
            />
          </div>,
          <div>
            <div
              className="side"
            >
              Phone Number:
            </div>
            <img
              className="phoneIcon"
              src="https://s3-us-west-1.amazonaws.com/imagesfornamson/telephone_318-61547.jpg"
            />
            <div
              className="sideInfoPhone"
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "moneyIcon",
            "src": "https://s3-us-west-1.amazonaws.com/imagesfornamson/21.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="side"
              >
                Price:
              </div>,
              <div
                className="sideInfoPrice"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Price:",
                "className": "side",
              },
              "ref": null,
              "rendered": "Price:",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "sideInfoPrice",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="side"
              >
                Hours of Operation:
              </div>,
              <img
                className="clockIcon"
                src="https://s3-us-west-1.amazonaws.com/imagesfornamson/clock.png"
              />,
              <div
                className="sideInfoHours"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Hours of Operation:",
                "className": "side",
              },
              "ref": null,
              "rendered": "Hours of Operation:",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "clockIcon",
                "src": "https://s3-us-west-1.amazonaws.com/imagesfornamson/clock.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "sideInfoHours",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="side"
              >
                Dress code:
              </div>,
              <img
                className="dressIcon"
                src="https://s3-us-west-1.amazonaws.com/imagesfornamson/img_473372.png"
              />,
              <div
                className="sideInfoStyle"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Dress code:",
                "className": "side",
              },
              "ref": null,
              "rendered": "Dress code:",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "dressIcon",
                "src": "https://s3-us-west-1.amazonaws.com/imagesfornamson/img_473372.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "sideInfoStyle",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="side"
              >
                Phone Number:
              </div>,
              <img
                className="phoneIcon"
                src="https://s3-us-west-1.amazonaws.com/imagesfornamson/telephone_318-61547.jpg"
              />,
              <div
                className="sideInfoPhone"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Phone Number:",
                "className": "side",
              },
              "ref": null,
              "rendered": "Phone Number:",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "phoneIcon",
                "src": "https://s3-us-west-1.amazonaws.com/imagesfornamson/telephone_318-61547.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "sideInfoPhone",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
